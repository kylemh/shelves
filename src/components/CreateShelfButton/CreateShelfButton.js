import React from 'react'
import PropTypes from 'prop-types';
import styles from './CreateShelfButton.module.scss';

export default class CreateShelfButton extends React.Component {
  static propTypes = {
    createShelf: PropTypes.func.isRequired,
  };

  render() {
    const { props } = this;

    return (
      <button className={styles.CreateShelfButton} onClick={props.createShelf}>
        <h2 className={styles.label}>+ Create New Shelf +</h2>
      </button>
    )
  }
}
