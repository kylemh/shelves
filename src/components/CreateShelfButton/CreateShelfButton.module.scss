@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';

.CreateShelfButton {
  background-color: rgba(0, 0, 0, 0);
  border: 4px dashed $feint;
  border-radius: 10px;
  height: $shelfHeight;
  width: 100%;
  transition: all 0.2s linear;

  .label {
    color: $feint;
  }

  @include on-hover-and-focus {
    cursor: pointer;
    background-color: $feint;
    border-style: solid;
    outline: none;

    .label {
      color: $primary;
    }
  }

  &:active {
    border-color: darken($color: $feint, $amount: 30);
  }
}
