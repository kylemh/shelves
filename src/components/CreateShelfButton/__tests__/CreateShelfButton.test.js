import React from 'react';
import { shallow } from 'enzyme';
import CreateShelfButton from '../CreateShelfButton';

describe('CreateShelfButton', () => {
  it('should render', () => {
    const wrapper = shallow(<CreateShelfButton createShelf={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call props.createShelf when clicked', () => {
    const createShelfFunction = jest.fn();
    const wrapper = shallow(<CreateShelfButton createShelf={createShelfFunction} />);

    expect(createShelfFunction).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(createShelfFunction).toHaveBeenCalledTimes(1);
  });
});
