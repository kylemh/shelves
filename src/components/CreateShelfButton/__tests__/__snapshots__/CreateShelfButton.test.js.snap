// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateShelfButton should render 1`] = `
<button
  className="CreateShelfButton"
  onClick={[MockFunction]}
>
  <h2
    className="label"
  >
    + Create New Shelf +
  </h2>
</button>
`;
